bandit/src/query-lambda/query-lambda.ts

```diff
@@ -1,41 +1,49 @@
+import type {SimpleQueryRowsResponse} from "@google-cloud/bigquery";
 import * as AWS from "aws-sdk";
 import { set, subHours } from "date-fns";
-import type { QueryExecution, Test } from "../lib/models";
-import { executeQuery } from "../lib/query";
-import { getQueries } from "./queries";
-
-const athena = new AWS.Athena({ region: "eu-west-1" });
+import type {  Test } from "../lib/models";
+import { buildAuthClient, getDataForBanditTest} from "./bigquery";
+import {buildWriteRequest, writeBatch} from "./dynamo";
+import {parseResultFromBigQuery} from "./parse-result";
+import {getSSMParam} from "./ssm";
 
 const stage = process.env.STAGE;
-const athenaOutputBucket = process.env.AthenaOutputBucket ?? "";
-const schemaName = "acquisition";
+const docClient = new AWS.DynamoDB.DocumentClient({ region: "eu-west-1" });
+
 
 export interface QueryLambdaInput {
 	tests: Test[];
 	date?: Date;
 }
 
-export async function run(input: QueryLambdaInput): Promise<QueryExecution[]> {
+export async function run(input: QueryLambdaInput): Promise<void> {
 	if (stage !== "CODE" && stage !== "PROD") {
 		return Promise.reject(`Invalid stage: ${stage ?? ""}`);
 	}
 
+	const ssmPath = `/bandit-testing/${stage}/gcp-wif-credentials-config`;
 	const date = input.date ?? new Date(Date.now());
 	const end = set(date, { minutes: 0, seconds: 0, milliseconds: 0 });
 	const start = subHours(end, 1);
+	const startTimestamp = start.toISOString().replace("T", " ");
+	const client = await getSSMParam(ssmPath).then(buildAuthClient);
 
-	const queries = getQueries(input.tests, stage, start, end);
-
-	const results: Array<Promise<QueryExecution>> = queries.map(
-		([test, query]) =>
-			executeQuery(query, athenaOutputBucket, schemaName, athena).then(
-				(executionId) => ({
-					executionId,
-					testName: test.name,
-					startTimestamp: start.toISOString(),
-				})
-			)
+	const resultsFromBigQuery: Array<{testName: string; rows: SimpleQueryRowsResponse}> = await Promise.all(
+		input.tests.map(test => getDataForBanditTest(client, stage, test, input.date))
 	);
 
-	return Promise.all(results);
+	const parsedResults = resultsFromBigQuery.map(({testName, rows}) => {
+		const parsed = parseResultFromBigQuery(rows);
+		return buildWriteRequest(parsed, testName, startTimestamp);
+	});
+	const batches = await Promise.all(parsedResults);
```

Comment:
there's no need for the `await Promise.all` here as we already awaited above

Reply:
In fact we don't need to reassign to `batches` at all here. The name batches is a bit misleading anyway, as we're just sending the whole array to dynamo